### PR TITLE
[CPython] Add platform macosx-arm64

### PR DESCRIPTION
**Pull request by HGuillemet**
Add support of apple silicon mac to cpython presets.

There are constants in global/python.java that map platform-specific configurations like `HAVE_*`. They'd better be skipped.

### CODE REVIEW

**Comment by saudet** [comment]
Please do the same with numpy and scipy

File: .github/actions/deploy-macosx/action.yml [hunk 1]
```diff
@@ -108,6 +108,11 @@ runs:
         export DYLD_LIBRARY_PATH="$LIBRARY_PATH"
         export DYLD_FALLBACK_LIBRARY_PATH="$LIBRARY_PATH"
 
+        # Change default JVM on macos-14 runner
+        if [[ -n ${JAVA_HOME_17_arm64:-} ]]; then
+          export JAVA_HOME=$JAVA_HOME_17_arm64
+        fi
+
```
Comment (saudet) [inline .github/actions/deploy-macosx/action.yml:115]: Java 8 or 11 isn't available??

**Comment by HGuillemet** [comment]
> Please do the same with numpy and scipy

I'm waiting for the merge of this one since numpy depends on cpython

File: .github/actions/deploy-macosx/action.yml (hunk 1 above)
Comment (HGuillemet) [reply .github/actions/deploy-macosx/action.yml:115]: > Java 8 or 11 isn't available??

In the github runner, 11 is, 8 is not.
We can install 8 from brew.
But I don't see the interest.

Comment (saudet) [reply .github/actions/deploy-macosx/action.yml:115]: I don't think Java 8 is available for arm64. Meh, Java 17 is fine I guess.

**Review by saudet — APPROVED** [review]

**Comment by saudet** [comment]
This breaks the build for at least OpenCV:
https://github.com/bytedeco/javacpp-presets/actions/runs/9632851195/job/26858888577
Could you fix that before trying to get SciPy working?

**Comment by HGuillemet** [comment]
> This breaks the build for at least OpenCV: https://github.com/bytedeco/javacpp-presets/actions/runs/9632851195/job/26858888577 Could you fix that before trying to get SciPy working?

The macosx-arm64 builds should now use the macos-14 runner, which is an arm64 machine.

**Comment by HGuillemet** [comment]
Do you want a separate PR for this change in opencv workflow ?

**Comment by saudet** [comment]
Yes, please!

**Pull request merged**